AutomatedTesting/Gem/PythonTests/scripting/ScriptEvents_AllParamDatatypes_CreationSuccess.py

```diff
@@ -44,163 +63,153 @@ def ScriptEvents_AllParamDatatypes_CreationSuccess():
+        node_tree_view = None
+        node_tree_search_frame = None
+        none_tree_search_box = None
+
+    def initialize_sc_qt_objects(self):
+        self.script_canvas = self.editor_window.findChild(QtWidgets.QDockWidget, SCRIPT_CANVAS_UI)
+        if self.script_canvas.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT) is None:
+            action = pyside_utils.find_child_by_pattern(self.script_canvas, {"text": NODE_PALETTE_UI, "type": QtWidgets.QAction})
             action.trigger()
-        node_palette = sc.findChild(QtWidgets.QDockWidget, "NodePalette")
-        tree = node_palette.findChild(QtWidgets.QTreeView, "treeView")
-        search_frame = node_palette.findChild(QtWidgets.QFrame, "searchFrame")
-        search_box = search_frame.findChild(QtWidgets.QLineEdit, "searchFilter")
-
-    def save_file():
-        editor.AssetEditorWidgetRequestsBus(bus.Broadcast, "SaveAssetAs", FILE_PATH)
-        action = pyside_utils.find_child_by_pattern(menu_bar, {"type": QtWidgets.QAction, "iconText": "Save"})
+        self.node_palette = self.script_canvas.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
+        self.node_tree_view = self.node_palette.findChild(QtWidgets.QTreeView, TREE_VIEW_QT)
+        self.node_tree_search_frame = self.node_palette.findChild(QtWidgets.QFrame, SEARCH_FRAME_QT)
+        self.none_tree_search_box = self.node_tree_search_frame.findChild(QtWidgets.QLineEdit, SEARCH_FILTER_QT)
+
+    def initialize_asset_editor_qt_objects(self):
+        self.editor_window = pyside_utils.get_editor_main_window()
+        self.asset_editor = self.editor_window.findChild(QtWidgets.QDockWidget, ASSET_EDITOR_UI)
+        self.asset_editor_widget = self.asset_editor.findChild(QtWidgets.QWidget, "AssetEditorWindowClass")
+        self.asset_editor_row_container = self.asset_editor_widget.findChild(QtWidgets.QWidget, "ContainerForRows")
+        self.menu_bar = self.asset_editor_widget.findChild(QtWidgets.QMenuBar)
+
+    def save_file(self):
+        editor.AssetEditorWidgetRequestsBus(bus.Broadcast, SAVE_ASSET_AS, FILE_PATH)
+        action = pyside_utils.find_child_by_pattern(self.menu_bar, {"type": QtWidgets.QAction, "iconText": SAVE_STRING})
         action.trigger()
         # wait till file is saved, to validate that check the text of QLabel at the bottom of the AssetEditor,
         # if there are no unsaved changes we will not have any * in the text
-        label = asset_editor.findChild(QtWidgets.QLabel, "textEdit")
-        return helper.wait_for_condition(lambda: "*" not in label.text(), 3.0)
+        label = self.asset_editor.findChild(QtWidgets.QLabel, "textEdit")
+        return helper.wait_for_condition(lambda: "*" not in label.text(), WAIT_TIME_3)
 
-    def expand_container_rows(object_name):
-        children = container.findChildren(QtWidgets.QFrame, object_name)
+    def expand_container_rows(self, object_name):
+        children = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, object_name)
```

Comment:
Most of these defs were used in the previous test

These should be a candidate for moving into a utils class in the next refactor of these tests. Please make a note of it. 
Unless you want to refactor these now.

Reply:
I like this idea.  Added a tools util file for these functions